app/Jobs/EncodeTrackFile.php

```diff
@@ -166,7 +172,29 @@ public function handle()
                     $this->trackFile->track->save();
                 }
 
-                File::delete($this->trackFile->track->getTemporarySourceFile());
+                if ($this->isReplacingTrack) {
+                    $oldVersion = $this->trackFile->track->current_version;
+
+                    // Update the version of the track being uploaded
+                    $this->trackFile->track->duration = \AudioCache::get($source)->getDuration();
```

Comment:
Stylistically, it's nicer to import `AudioCache` and use it here without a leading backslash.